src/main/scala/com/ironcorelabs/DeviceContext.scala

```diff
@@ -7,21 +7,40 @@ import com.ironcorelabs.{sdk => jsdk}
 /**
  * Account's device context. Needed to initialize the SDK with a set of device keys.
  *
- * @param id ID of a user
+ * @param deviceId ID of a device
+ * @param userId ID of a user
  * @param segmentId ID of a segment
- * @param privateDeviceKey asymmetric private key
- * @param deviceSigningKeyPair signing keypair specific to a device
+ * @param devicePrivateKey asymmetric private key
+ * @param signingPrivateKey signing keypair specific to a device
  */
 case class DeviceContext(
-  id: UserId,
+  deviceId: DeviceId,
+  userId: UserId,
   segmentId: Long,
-  privateDeviceKey: PrivateKey,
-  deviceSigningKeyPair: DeviceSigningKeyPair
+  devicePrivateKey: PrivateKey,
+  signingPrivateKey: DeviceSigningKeyPair
 ) {
   private[sdk] def toJava[F[_]](implicit syncF: Sync[F]): F[jsdk.DeviceContext] =
     for {
-      javaId  <- id.toJava
-      javaPK  <- privateDeviceKey.toJava
-      javaSKP <- deviceSigningKeyPair.toJava
-    } yield new jsdk.DeviceContext(javaId, segmentId, javaPK, javaSKP)
+      javaDeviceId <- deviceId.toJava
+      javaUserId   <- userId.toJava
+      javaDPK      <- devicePrivateKey.toJava
+      javaSPK      <- signingPrivateKey.toJava
+    } yield new jsdk.DeviceContext(javaDeviceId, javaUserId, segmentId, javaDPK, javaSPK)
+
+  def toJsonString[F[_]](implicit syncF: Sync[F]): F[String] =
+    toJava[F].map(_.toJsonString)
+}
+
+object DeviceContext {
+  def fromJsonString(jsonString: String): DeviceContext = {
+    val javaContext = jsdk.DeviceContext.fromJsonString(jsonString)
+    DeviceContext(
+      DeviceId(javaContext.getDeviceId.getId),
+      UserId(javaContext.getAccountId.getId),
+      javaContext.getSegmentId,
```

Comment:
Do I want to know why this one doesn't need a `.getId` on it?

Reply:
https://github.com/IronCoreLabs/ironoxide/issues/25

Reply:
`getSegmentId` directly returns a Long